dynawo/sources/Models/CPP/Events/ModelAreaShedding/DYNModelAreaShedding.h

```diff
@@ -0,0 +1,295 @@
+//
+// Copyright (c) 2022, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file  DYNModelAreaShedding.h
+ *
+ * @brief
+ *
+ */
+#ifndef MODELS_CPP_EVENTS_MODELAREASHEDDING_DYNMODELAREASHEDDING_H_
+#define MODELS_CPP_EVENTS_MODELAREASHEDDING_DYNMODELAREASHEDDING_H_
+
+#include "DYNModelCPP.h"
+#include "DYNSubModelFactory.h"
+
+namespace DYN {
+
+/**
+ * @brief ModelAreaSheddingFactory Model factory
+ *
+ * Implementation of @p SubModelFactory template for ModelAreaSheddingFactory Model
+ */
+class ModelAreaSheddingFactory : public SubModelFactory {
+ public:
+  /**
+   * @brief default constructor
+   *
+   */
+  ModelAreaSheddingFactory() { }
+  /**
+   * @brief default destructor
+   *
+   */
+  ~ModelAreaSheddingFactory() { }
+
+  /**
+   * @brief Model AreaShedding getter
+   *
+   * @return A pointer to a new instance of Model AreaShedding
+   */
+  SubModel* create() const;
+
+  /**
+   * @brief Model AreaShedding destroy
+   */
+  void destroy(SubModel*) const;
+};
+
+/**
+ * class ModelAreaShedding
+ */
+class ModelAreaShedding : public ModelCPP {
+ public:
+  /**
+   * @brief define type of calculated variables
+   *
+   */
+  typedef enum {
+    nbCalculatedVars_ = 0
+  } CalculatedVars_t;
+
+  /**
+   * @brief Default constructor
+   *
+   * Creates a new ModelAreaShedding instance.
+   */
+  ModelAreaShedding();
+  /**
+   * @brief Default destructor
+   *
+   * Creates a new ModelAreaShedding instance.
```

Comment:
Wrong comment

Reply:
I don't understand what is the expected correction?

Reply:
The default destructor "a priori" doesn't create an instance :)

Reply:
solved